includes/classes/Customer.php

```diff
@@ -645,6 +645,30 @@ public function getNumberOfOrders(): int
         return $result->fields['total'];
     }
 
+    /**
+     * Return a simple summary of orders for this customer, ordered most recent first.
+     *
+     * @param integer $limit Optional limit of the number of orders to return.
+     * @return array
+     */
+    public function getOrdersSummary(int $limit = 0): array {
+        global $db;
+        if (empty($this->customer_id)) {
+            return [];
+        }
+        $rs = $db->Execute("SELECT orders_id, date_purchased, orders_status_name
+            FROM " . TABLE_ORDERS . " o
+            INNER JOIN " . TABLE_ORDERS_STATUS . " os
+                ON o.orders_status = os.orders_status_id AND os.language_id = {$_SESSION['languages_id']}
+            WHERE customers_id = {$this->customer_id}
+            ORDER BY date_purchased DESC" . (!empty($limit) ? " LIMIT $limit" : '') . ';');
+        $result = [];
+        foreach ($rs as $row) {
+            $result[] = $row;
+        }
+        return $result;
```

Comment:
Since this is returning an array of db records, there's no need to foreach over it. You can foreach over the results of the db query directly, so could return that directly.

Also, and sorry if this is super nitpicky: 
Instead of `$rs` for a variable, in Zen Cart we commonly use `$result`, since it is the query's result.
And then if a temporary value is needed for parsing results to be returned, commonly you'll see `$retVal` or `$response` used.

Reply:
My aim was to avoid returning a database object (`mysqli_result`? not sure) from a data accessor, figured a plain array would be more standard.  Looks like I will squash this new method anyway, though, if I can get `getOrderHistory()` to play nice with `splitPageResult`. If not, will fix this as you suggest.